doc/documentation-main.html

```diff
@@ -2343,6 +2343,45 @@ <h3 class="sourcetitle">Shell</h3>
 }
 </pre>
 
+<p>
+<!-------------------------------------
+  IALTERSUITELISTENER
+  ------------------------------------>
+
+<h4><a class="section" indent=".." name="ialtersuite">Altering suites (or) tests</a></h4>
+Sometimes we may need to just want to alter a suite (or) a test tag in a suite xml in runtime without having to
+change the contents of a suite file.<br>
+
+A classic example for this would be to try and leverage your existing suite file and try using it for simulating a load test
+on your "Application under test".
+If you were to do this on your own, at the minimum you would end up duplicating the contents of your &lt;test&gt; tag multiple
+times and create a new suite xml file and work with. But this doesn't seem to scale a lot.
+
+TestNG allows you to alter a suite (or) a test tag in your suite xml file at runtime via listeners.
+You achieve this by providing a listener that implements <a href="../javadocs/org/testng/IAlterSuiteListener.html"><tt>IAlterSuiteListener</tt></a>.
+Please refer to <a href="#testng-listeners">Listeners section</a> to learn about listeners.
+<p>
+Here is an example that shows how the suite name is getting altered in runtime:
+
+<pre class="brush: java">
+public class AlterSuiteNameListener implements IAlterSuiteListener {
+
+    @Override
+    public void alter(List&lt;XmlSuite&gt; suites) {
+        XmlSuite suite = suites.get(0);
+        suite.setName(getClass().getSimpleName());
+    }
+}
+</pre>
+
+This listener can only be added to your execution via the following two ways :
```

Comment:
"This listener can only be added with either of the following ways" (your phrasing makes it sound you need to do both).